### ISSUE TITLE
Hostfolder mount permissions cannot be changed from 755

### ISSUE THREAD

**Issue by riddins** [minishift/minishift#2157]
### General information

  * Minishift version: v1.14.0+1ec5877
  * OS: macOS
  * Hypervisor: xhyve 

### Steps to reproduce

  1. minishift hostfolder add 
  2. minishift hostfolder mount
  3. minishift ssh "ls -la /path/to/mount"
  4. minishift ssh "sudo chmod -R 777 /path/to/mount"
  5. minishift ssh "ls -la /path/to/mount"

### Expected

permissions are changed to 777

### Actual

permissions are unchanged and remain 755

### More Info

A workaround for this issue is to not use minishift hostfolder command and mount the cifs share manually.  Manually mounting with the following command as root on the minishift vm works as expected:

mount -t cifs //host/share /path/to/mount username=share_user, password=share_pass, uid=minishift_user, gid=minishift_group, rw, file_mode=0777, dir_mode=0777

**Comment by gbraad** [comment]
it should be possible to add this as additional parameters for the mount of cifs and sshfs.

**Comment by anjannath** [comment]
@riddins So this is only happening for CIFS share, could you please try the SSHFS hostfolder mount and see if you face the issue? 
https://docs.openshift.org/latest/minishift/using/host-folders.html#sshfs-2
SSHFS is going to be our default for _host folder mount_ from the next release.

**Comment by anjannath** [comment]
> it should be possible to add this as additional parameters for the mount of cifs and sshfs.

we could add `file_mode` and `dir_mode` as additional parameters for CIFS and `umask` for SSHFS
but i guess you are talking about having a generic option (abstracted) that would work for both. while mounting with SSHFS i can change the permission using `chmod`

Also, changing the permissions using `chmod` not working, is probably dependent on the CIFS server.. below snippet is from the `mount.cifs` manpage.. 
```
file_mode=arg
              If the server does not support the CIFS Unix extensions this overrides the default file mode.
dir_mode=arg
              If the server does not support the CIFS Unix extensions this overrides the default mode for directories.

```

**Comment by anjannath** [comment]
@riddins you can pass the extra options using the `--options` flag for `hostfolder add` command.
Please give it a try with binary from http://artifacts.ci.centos.org/minishift/minishift/pr/2508/

`minishift hostfolder add --type cifs --source //host/share --target /path/to/mount --options "username=share_user, password=share_pass, uid=minishift_user, gid=minishift_group, rw, file_mode=0777, dir_mode=0777"`

**Comment by LalatenduMohanty** [comment]
Resolved via #2508
